Pure_Parallels/update_parallel_wcs.py

```diff
@@ -0,0 +1,155 @@
+import os, sys
+import shutil
+import glob
+from astropy.io import ascii
+import astropy.io.fits as pyfits
+import numpy as np
+
+def log_comment(LOGFILE, comment, verbose=False, show_date=False, mode='a'):
+    """
+    Log a message to a file, optionally including a date tag
+    """
+    import time
+
+    if show_date:
+        msg = '# ({0})\n'.format(nowtime())
+    else:
+        msg = ''
+
+    msg += '{0}\n'.format(comment)
+
+    if LOGFILE is not None:
+        fp = open(LOGFILE, mode)
+        fp.write(msg)
+        fp.close()
+
+    if verbose:
+        print(msg[:-1])
+
+    return msg
+
+def update_pure_parallel_wcs(file, logfile="pure_parallel_wcs_logfile",
+                             fix_vtype='PARALLEL_PURE', verbose=True):
+    """
+    Update pointing-related keywords of pure parallel exposures using the
+    pointing info from the FGS (and the prime exposures) from the MAST database
+    and `pysiaf`
+    
+    1. Find the FGS log from a MAST query that starts before the pure parallel
+       exposure starts and ends after the exposure ends.
+    2. Use the ``ra_v1, dec_v1, pa_v3`` values from the FGS log to set the
+       pointing attitude with `pysiaf`.
+    3. Compute the sky position of the ``CRPIX`` reference pixel of ``file``
+       with `pysiaf` and put that position in the ``CRVAL`` keywords.
+    
+    Parameters
+    ----------
+    file : str
+        Filename of a pure-parallel exposure (typically a rate.fits file)
+    
+    fix_vtype : str
+        Only run if ``file[0].header['VISITYPE'] == fix_vtype``
+    
+    verbose : bool
+        Status messaging
+    
+    Returns
+    -------
+    status : None, True
+        Returns None if some problem is found 
+    
+    """
+    import pysiaf    
```

Comment:
Again imports should be located at top of file (PEP8 - standards).

Reply:
done